go/packages/slslambda/event.go

```diff
@@ -16,19 +18,37 @@ const (
 )
 
 type (
+	EventOptions struct {
+		CustomFingerprint *string
+		CustomTags        map[string]string
+	}
 	errorEvent struct {
 		timestamp time.Time
 		error
+		*EventOptions
 	}
 	warningEvent struct {
 		timestamp time.Time
 		message   string
+		*EventOptions
 	}
 	protoEvent interface {
 		ToProto(traceID, parentSpanID []byte) (*instrumentationv1.Event, error)
 	}
 )
 
+func marshalCustomTags(options *EventOptions) *string {
+	if options.CustomTags != nil && len(options.CustomTags) > 0 {
+		b, err := json.Marshal(options.CustomTags)
```

Comment:
```suggestion
	if options.CustomTags == nil || len(options.CustomTags) = 0 {
	  return nil
	}
		b, err := json.Marshal(options.CustomTags)
```

It's better to return early to reduce nesting https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_return_early_rather_than_nesting_deeply